zipkin-ui/README.md

```diff
@@ -0,0 +1,34 @@
+# zipkin-ui
+
+Zipkin-UI is a single-page application that reads configuration from `/config.json`.
+
+When looking at a trace, the browser is sent to the path `/traces/{id}`. For the single-page
+app to serve that route, the server needs to forward the request to `index.html`. The same
+forwarding applies to `/dependencies` and any other routes the UI controls. 
+
+Under the scenes the JavaScript code looks at `window.location` to figure out what the
+UI should do. This is handled by a route api defined in the crossroads library.
+
+The full list of
+such routes is defined in [`resources/zipkin-ui/index-routes.json`](src/main/resources/zipkin-ui/index-routes.json).
+Routes ending with `/` should be used in prefix matching (`/foo/` should match `/foo/`, `/foo/bar`, and `/foo/bar/baz`).
+Routes not ending with `/` should be used in exact matching (`/foo` should match `/foo` and nothing else).
+
+Based on all these, there are several ways for serving zipkin-ui:
+
+ * Programatically read `index-routes.json` and set up explicit routes based on it. Example: [zipkin-web](https://github.com/openzipkin/zipkin/blob/5dec252e4c562b21bac5ac2f9d0b437d90988f79/zipkin-web/src/main/scala/com/twitter/zipkin/web/Main.scala#L92-L95).
```

Comment:
I'd rather not maintain index-routes.json... I firmly believe that serving index.html as a fallback for every route that has no other route handler should be a basic feature of any web server; if it's not, then it should be that server's responsibility to bear the burden of finding a workaround.
(I'm sorry that I'm so stubborn)